source/common/config/grpc_mux_impl.cc

```diff
@@ -17,7 +17,11 @@ GrpcMuxImpl::GrpcMuxImpl(const LocalInfo::LocalInfo& local_info, Grpc::AsyncClie
       service_method_(service_method), random_(random), time_source_(dispatcher.timeSystem()),
       control_plane_stats_(generateControlPlaneStats(scope)) {
   Config::Utility::checkLocalInfo("ads", local_info);
+
+  // Bucket contains 100 tokens maximum and refills at 10 tokens/sec.
+  limit_request_ = std::make_unique<TokenBucketImpl>(MAX_TOKENS, time_source_, TOKEN_REFILL_RATE);
```

Comment:
@junr03 I do think we want to force people (with ability to tune those limits) in to rate limiting for overall benefit of control plane and envoy. I am not sure if any one wants to disable this unless there is a bug and want to get the old behaviour. I think providing disabling option may not be a good idea which will eventually be controlled by configured limits. But if you feel strongly about it  i can make these limits configurable now it self.
@htuch WDYT?

Reply:
I also wonder if the default limits (100 tokens and 10 tokens/sec) we have would be sufficient not to get in to rate limit for most cases. I think only EDS updates might get in to rate limits in very specific cases?

Reply:
@ramaraochavali yeah, we should probably add this as it's pretty easy. Your proto makes sense I think, do you want to put it in `GrpcService` as it only applies there?

Reply:
@htuch I thought about putting in `GrpcService` but since these settings are not generically applicable all `GrpcService`'s for example external autz, rate limiting etc I preferred, `ApiConfigSource`. WDYT?